program/include/rcmail_action.php

```diff
@@ -151,6 +151,26 @@ public static function table_output($attrib, $table_data, $show_cols, $id_col)
         return $table->show($attrib);
     }
 
+    /**
+     * Set env-flag for quota. It is used in templates to determine if quota
+     * data should be displayed.
+     * In case the quota is zero and the config option
+     * `quota_zero_as_unlimited` is true, then the account has no quota and
+     * should not be shown a meaningless (or even wrong) quota-display.
+     */
+    public static function quota_set_env()
+    {
+        $rcmail = rcmail::get_instance();
+        // This call also checks the QUOTA capability of the server, we don't
+        // need to do that manually (the return value is `false` if the server
+        // doesn't support it).
+        $quota = $rcmail->storage->get_quota();
+        $zero_as_unlimited = (bool) $rcmail->config->get('quota_zero_as_unlimited');
```

Comment:
since you have a config param here, you could check that first and early return if that config param is set to false.

```suggestion
        $zero_as_unlimited = (bool) $rcmail->config->get('quota_zero_as_unlimited');
        
        if($zero_as_unlimited === false) {
        	$rcmail->output->set_env('quota', true);
        	return;
        }
```

Reply:
Thank you for the hint! I added a second early check, which catches the case that the server doesn't event support quotas.